examples/vpc/main.tf

```diff
@@ -41,36 +50,181 @@ resource "aws_route_table" "test" {
   tags = var.tags
 }
 
-resource "aws_route_table_association" "test" {
-  subnet_id      = aws_subnet.test.id
+resource "aws_route_table_association" "client" {
+  subnet_id      = aws_subnet.client.id
   route_table_id = aws_route_table.test.id
 }
 
-module "redpanda-cluster" {
-  source                   = "../../"
-  public_key_path        = var.public_key_path
-  nodes                  = var.nodes
-  deployment_prefix      = var.deployment_prefix
-  enable_monitoring      = var.enable_monitoring
-  tiered_storage_enabled = var.tiered_storage_enabled
-  allow_force_destroy    = var.allow_force_destroy
-  vpc_id                 = aws_vpc.test.id
-  distro                 = var.distro
-  hosts_file             = var.hosts_file
-  tags                   = var.tags
-  subnet_id = aws_subnet.test.id
-  availability_zone = ["us-west-2a"]
+resource "aws_route_table_association" "server" {
+  subnet_id      = aws_subnet.server.id
+  route_table_id = aws_route_table.test.id
 }
 
+resource "aws_security_group" "client_sec_group" {
+  name        = "${var.deployment_prefix}-client-sg"
+  description = "client security group"
+  tags        = var.tags
+  vpc_id      = aws_vpc.test.id
+  ingress {
+    description = "Allow oubound to ssh"
+    from_port   = 22
+    to_port     = 22
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
 
-resource "aws_route53_record" "private_record" {
-  count = var.nodes
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+}
 
-  zone_id = aws_route53_zone.test.zone_id
-  name    = "${element(keys(module.redpanda-cluster.redpanda_map), count.index)}.local"
-  type    = "A"
-  ttl     = "300"
-  records = [element(values(module.redpanda-cluster.redpanda_map), count.index)]
+module "redpanda-cluster" {
+  source                          = "../../"
+  public_key_path                 = var.public_key_path
+  nodes                           = var.nodes
+  deployment_prefix               = var.deployment_prefix
+  enable_monitoring               = var.enable_monitoring
+  tiered_storage_enabled          = var.tiered_storage_enabled
+  allow_force_destroy             = var.allow_force_destroy
+  vpc_id                          = aws_vpc.test.id
```

Comment:
Shouldn't this point to the variable?:

```
variable "vpc_id" {
  description = "only set when you are planning to provide your own network rather than using the default one"
  type        = string
  default     = ""
}
```

Reply:
Not in this instance as we're actually creating the VPC in this one. The variable should probably be deleted.